### PR TITLE
integrate tribuo anomaly detection based on libSVM

### LINKED ISSUE #98: Add anomaly localization function
Anomaly localization is to find the top contributors to some abnormal period like which server contributes most for the error spike.
We can integrate anomaly localization with ml-commons to provide general anomaly localization function.

### PR DESCRIPTION
**Pull request by ylwu-amzn**
Signed-off-by: Yaliang Wu <ylwu@amazon.com>

### Description
Integrate tribuo anomaly detection algorithm which based on libSVM. Check this doc https://tribuo.org/learn/4.1/docs/features.html#anomaly-detection
 
### Issues Resolved
#98 
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/ad/AnomalyDetection.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        if (parameters.getEpsilon() != null) {
+            params.setEpsilon(parameters.getEpsilon());
+        }
+        if (parameters.getDegree() != null) {
+            params.setDegree(parameters.getDegree());
+        }
+        MutableDataset<Event> data = TribuoUtil.generateDataset(dataFrame, new AnomalyFactory(),
+                "Anomaly detection LibSVM training data from OpenSearch", TribuoOutputType.ANOMALY_DETECTION_LIBSVM);
+
+        LibSVMAnomalyTrainer trainer = new LibSVMAnomalyTrainer(params);
```
Comment (Zhangxunmt) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/ad/AnomalyDetection.java:117]: nitpick: why not use a function to instantiate the params and LibSVMAnomalyTrainer to make the train() body smaller?

File: common/src/main/java/org/opensearch/ml/common/parameter/AnomalyDetectionParams.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        }
+        if (degree != null) {
+            builder.field(DEGREE_FIELD, degree);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public int getVersion() {
```
Comment (Zhangxunmt) [inline common/src/main/java/org/opensearch/ml/common/parameter/AnomalyDetectionParams.java:166]: Just curious: this version is always 1?

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/ad/AnomalyDetection.java (hunk 1 above)
Comment (Zhangxunmt) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/ad/AnomalyDetection.java:117]: nitpick: why not use a function to instantiate the params and LibSVMAnomalyTrainer to make the train() body smaller?

**Review by Zhangxunmt — DISMISSED** [review]
approved with 2 comments.

File: common/src/main/java/org/opensearch/ml/common/parameter/AnomalyDetectionParams.java (hunk 2 above)
Comment (ylwu-amzn) [reply common/src/main/java/org/opensearch/ml/common/parameter/AnomalyDetectionParams.java:166]: We set version as 1 for first release. May change to higher version in future

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/ad/AnomalyDetection.java (hunk 1 above)
Comment (ylwu-amzn) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/ad/AnomalyDetection.java:117]: Will tune this part in next PR.

**Review by Zhangxunmt — APPROVED** [review]

**Review by jackiehanyang — APPROVED** [review]

**Pull request merged**